### PR TITLE
fossid-webapp: Compatibility with FossID 2021.2.3

### PR DESCRIPTION
**Pull request by nnobelis**
Adapt the FossID scanner to be compatible with the latest version of
FossID. Compatibility with previous versions is preserved.
The only changes impacting ORT are the ones made to the [Project] model
class.

List of relevant changes:
projects - get_information
  * Response no longer has the keys 0 - 14 included that duplicates the
  information of the named keys.
  * Response datatype of project id changed from string to integer
  * Response datatype of creator value changed from string to integer
  * Response datatype of is_archived value changed from string to
  integer

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

### CODE REVIEW

**Comment by nnobelis** [comment]
Just a sidenote: this change is not mandatory because our current codebase is already compatible with FossID 2021.2.3.
However, the changes in this PR allow to have a 100% correct representation of the entities returned by this FossID version.

File: clients/fossid-webapp/src/main/kotlin/VersionedFossIdService2021dot2.kt [hunk 1]
```diff
@@ -31,4 +32,12 @@ class VersionedFossIdService2021dot2(
             delegate.checkScanStatus2021dot2(
                 PostRequestBody("check_status", SCAN_GROUP, user, apiKey, "scan_code" to scanCode)
             )
+    override suspend fun getProject(
```
Comment (sschuberth) [inline clients/fossid-webapp/src/main/kotlin/VersionedFossIdService2021dot2.kt:35]: Please add a blank line before this one.

File: clients/fossid-webapp/src/main/kotlin/FossIdServiceWithVersion.kt [hunk 2]
```diff
@@ -49,4 +51,12 @@ abstract class FossIdServiceWithVersion(val version: String) : FossIdRestService
...
         scanCode: String
     ): EntityResponseBody<out UnversionedScanDescription>
+
+    /**
+     * Get the project status for the given [projectCode].
+     *
+     * The HTTP request is sent with [user] and [apiKey] as credentials.
+     */
+    abstract suspend fun getProject(user: String, apiKey: String, projectCode: String
+    ): EntityResponseBody<out UnversionedProject>
```
Comment (sschuberth) [inline clients/fossid-webapp/src/main/kotlin/FossIdServiceWithVersion.kt:61]: Broken formatting.

File: clients/fossid-webapp/src/main/kotlin/model/UnversionedProject.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.fossid.model
+
+/**
+ * An interface abstracting a Project, regardless of the version of the FossID server.
+ */
+interface UnversionedProject {
```
Comment (sschuberth) [inline clients/fossid-webapp/src/main/kotlin/model/UnversionedProject.kt:25]: Could you please separate out some independent preparations, like the introduction of this interface, to preceding commits?

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Comment by nnobelis** [comment]
DO NOT MERGE: FossID are rolling back some of their changes.
Let's wait a little what is coming out from the discussions between FossID and the new version early adopters.

**Comment by sschuberth** [comment]
I'm moving this back to draft state to signal this should not yet be merged.

**Pull request closed without merging**